### PR TITLE
refactor: clean up cmake

### PR DESCRIPTION
**Pull request by gocarlos**

### CODE REVIEW

File: conanfile.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# conan usage:
```
Comment (gocarlos) [inline conanfile.txt:1]: this is optional, for people using modern tools like the conan package manager...

File: scripts/format_code.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
```
Comment (gocarlos) [inline scripts/format_code.sh:1]: this is helpful to keep things clean, previous the style was a bit heterogeneous

**Comment by gocarlos** [comment]
@jimsch please review, this should be the last cmake MR 👯

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,202 +1,224 @@
...
+                 -DUSE_SHARED_MBEDTLS_LIBRARY=${BUILD_SHARED_LIBS}
+                 -DENABLE_TESTING=OFF
+                 -DLIB_INSTALL_DIR=${install_dir}/lib
+                 INTALL_DIR
+                 "${dist_dir}"
+      UPDATE_DISCONNECTED 1)
+
+    externalproject_get_property(project_mbedtls install_dir)
+    include_directories("${install_dir}/include")
+    add_library(mbedtls STATIC IMPORTED)
```
Comment (jimsch) [inline CMakeLists.txt:207]: You kept this line as common here, but you made it dependent on MSVC for cn-cbor.  Was there a reason?

File: scripts/format_code.sh (hunk 2 above)
Comment (jimsch) [reply scripts/format_code.sh:1]: Is this just touching the CMake files right now and not the source code files?

Comment (gocarlos) [reply scripts/format_code.sh:1]: Correct

File: CMakeLists.txt (hunk 3 above)
Comment (gocarlos) [reply CMakeLists.txt:207]: Do not get it, cam you explain?

Comment (jimsch) [reply CMakeLists.txt:207]: Lines 170 and 174 do "add_library(cn-cbor STATIC IMPORTED)"  Once for each branch of the "if (MSVC)" but here you are doing it commonly outside of the "if"

Comment (gocarlos) [reply CMakeLists.txt:207]: Aha, yeah, we can do the same for cn-cbor

Comment (gocarlos) [reply CMakeLists.txt:207]: It was already like this, see line l 161 of diff above

**Pull request merged**